clang/lib/Basic/DiagnosticIDs.cpp

```diff
@@ -599,6 +600,19 @@ DiagnosticIDs::getDiagnosticSeverity(unsigned DiagID, SourceLocation Loc,
       SM.isInSystemMacro(Loc))
     return diag::Severity::Ignored;
 
+  // Apply suppression mappings if severity wasn't explicitly mapped with a
+  // clang-diagnostics pragma to ensure pragmas always take precedence over
```

Comment:
What happens when a pragma explicitly enables a warning?

Reply:
pragma takes precedence and we don't suppress via the mappings. do you think the comment needs rewording?

Reply:
Carefully reading the comment, it's clear.
We could make it easier perhaps.

// Clang-diagnostics pragmas always take precedence over suppression mapping.

The rest of the information in the comment can be put inside the first if().